packages/core/src/providers/network/provider.tsx

```diff
@@ -1,14 +1,30 @@
-import { ReactNode, useCallback, useReducer } from 'react'
+import { ReactNode, useCallback, useReducer, useState } from 'react'
 import { NetworkContext } from './context'
 import { defaultNetworkState, networksReducer } from './reducer'
 import { Network } from './model'
+import { JsonRpcProvider, Web3Provider } from '@ethersproject/providers'
+import { EventEmitter } from 'events'
+import { subscribeToProviderEvents } from '../../helpers/eip1193'
 
 interface NetworkProviderProps {
   children: ReactNode
 }
 
+async function tryToGetAccount(provider: JsonRpcProvider) {
+  try {
+    return await provider.getSigner().getAddress()
+  } catch (e) {
+    if (e.code === 'UNSUPPORTED_OPERATION') {
+      // readonly provider
+      return undefined
+    }
+    throw e
+  }
+}
```

Comment:
👍

Reply:
Ha ha, I have a next question for you about error handling: Should we throw error, or add error to state?

Reply:
It will call reportError after it's caught in `activate` function. Btw this is still not obvious if we want it - a fail fast approach might work better here and we should throw when somebody tries to activate bad provider